internal/rest/client.go

```diff
@@ -340,6 +340,20 @@ func (c *Client) LastError() error {
 	return c.lastErr
 }
 
+// computes the exponential backoff duration according to
+// https://www.awsarchitectureblog.com/2015/03/backoff.html
+func exponentialBackoffWait(r *rand.Rand, unit, cap time.Duration) func(uint) time.Duration {
+	return func(attempt uint) time.Duration {
+		// sleep = random_between(0, min(cap, base * 2 ** attempt))
+		sleep := unit * time.Duration(1<<attempt)
+		if sleep > cap {
+			sleep = cap
+		}
+		sleep -= time.Duration(r.Float64() * float64(sleep))
```

Comment:
Any reason why you chose the above formula for `sleep` instead of
```go
sleep = time.Duration(r.Float64() * float64(sleep)
```

Reply:
```suggestion
		sleep -= time.Duration(r.Float64() * float64(sleep-unit))
```
... at least with this, it makes sense (and we always sleep at least "unit"

Reply:
@krisis the point here is to specify a large time frame in each retry then each server will pick a random time in that.